From 7a2feeac4822c614d9092a6ca0ed968eebaeb17d Mon Sep 17 00:00:00 2001
From: beviu <contact@beviu.com>
Date: Mon, 5 May 2025 17:39:40 +0200
Subject: [PATCH] x86/uintr: Set UINV to X86_TRAP_PF for testing

This patch is used to test whether setting the UINV to X86_TRAP_PF
allows us to receive a user interrupt when a page fault occurs.

This is just for debugging, do not use on a production kernel!
---
 arch/x86/kernel/uintr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/uintr.c b/arch/x86/kernel/uintr.c
index ecef63974f68..bcb809555056 100644
--- a/arch/x86/kernel/uintr.c
+++ b/arch/x86/kernel/uintr.c
@@ -986,7 +986,7 @@ static int do_uintr_register_handler(u64 handler, unsigned int flags)
 
 	/* Modify only the relevant bits of the MISC MSR */
 	xsave_rdmsrl(xstate, MSR_IA32_UINTR_MISC, &misc_msr);
-	misc_msr |= (u64)UINTR_NOTIFICATION_VECTOR << 32;
+	misc_msr |= (u64)X86_TRAP_PF << 32;
 	xsave_wrmsrl(xstate, MSR_IA32_UINTR_MISC, misc_msr);
 
 	t->thread.upid_activated = true;
-- 
2.49.0

